### PR TITLE
Update crate edition to 2021

### LINKED ISSUE #2249: Update to Rust 2021
We should update our crates to rust 2021. This should mostly be doable via `rustfix` / `cargo fix`.

### PR DESCRIPTION
**Pull request by thomcc**
This was bugging me, and is an easy update. I'm not sure if we need to update the Rust version in Docker (seems possible).

Fixes #2249

### CODE REVIEW

**Comment by mrinalwadhwa** [comment]
@metaclips can help upgrade the builder image.
Checks pass without upgrading the image.

**Comment by thomcc** [comment]
Yeah, I don't think we ended up needing to update the image.

File: implementations/rust/ockam/ockam_examples/example_projects/credentials/src/bin/bob.rs [hunk 1]
```diff
@@ -7,7 +7,6 @@ use ockam::{
     CredentialProtocol, Identity, Identity, IdentityIdentifier, Result, TcpTransport,
     TrustIdentifierPolicy, Vault, TCP,
 };
-use std::convert::TryFrom;
```
Comment (antoinevg) [inline implementations/rust/ockam/ockam_examples/example_projects/credentials/src/bin/bob.rs:10]: This makes me happy.

File: implementations/rust/ockam/ockam_transport_core/src/error_test.rs [hunk 2]
```diff
@@ -19,7 +18,8 @@ fn code_and_domain() {
         (11, TransportError::Encoding),
         (12, TransportError::Protocol),
         (13, TransportError::GenericIo),
-    ])
+    ]
+    .into_iter()
```
Comment (antoinevg) [inline implementations/rust/ockam/ockam_transport_core/src/error_test.rs:22]: Yay!

**Review by antoinevg — APPROVED** [review]
Looking good!

File: implementations/rust/ockam/ockam_examples/example_projects/credentials/src/bin/bob.rs (hunk 1 above)
Comment (thomcc) [reply implementations/rust/ockam/ockam_examples/example_projects/credentials/src/bin/bob.rs:10]: Definitely same.

**Pull request merged**